### ISSUE TITLE
Improve `Style/CaseEquality` cop

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12355]
## Description

Avoid explicit use of the case equality operator `===`. As its name implies it is meant to be used implicitly by case expressions and outside of them it yields some pretty confusing code.

https://rubystyle.guide/#no-case-equality

## Describe the solution you'd like

There is no autocorrection for the following example:

```ruby
# bad
/something/ === some_string

# good
some_string.match?(/something/)
```

**Comment by koic** [comment]
It is already detected:

```console
$ echo "/something/ === some_string" | bundle exec rubocop --stdin test.rb -A --only Style/CaseEquality
Inspecting 1 file
C

Offenses:

test.rb:1:13: C: Style/CaseEquality: Avoid the use of the case equality operator ===.
/something/ === some_string
            ^^^

1 file inspected, 1 offense detected
====================
/something/ === some_string
```

Autocorrect is not supported because `some_string` is not compatible when it's `nil`:

```ruby
/regexp/.match?(nil) #=> false
nil.match?(/regexp/) #=> NoMethodError
```

**Comment by ydakuka** [comment]
Yes that's right, but if I take another example, the autocorrection will be done.


```ruby
ydakuka@yauhenid:~/Work/project$ echo "Array === something" | bundle exec rubocop --stdin test.rb -A --only Style/CaseEquality
Inspecting 1 file
C

Offenses:

test.rb:1:7: C: [Corrected] Style/CaseEquality: Avoid the use of the case equality operator ===.
Array === something
      ^^^

1 file inspected, 1 offense detected, 1 offense corrected
====================
something.is_a?(Array)
```
